src/core/server/saved_objects/service/lib/point_in_time_finder.ts

```diff
@@ -63,17 +71,20 @@ export interface ISavedObjectsPointInTimeFinder {
 /**
  * @internal
  */
-export class PointInTimeFinder implements ISavedObjectsPointInTimeFinder {
+export class PointInTimeFinder<T = unknown, A = unknown>
+  implements ISavedObjectsPointInTimeFinder<T, A> {
   readonly #log: Logger;
   readonly #client: PointInTimeFinderClient;
   readonly #findOptions: SavedObjectsFindOptions;
```

Comment:
**Why I used the PIT finder:** these searches could potentially return thousands of results. Realistically I don't think we would hit the 10k limit, but we could, and I also wanted to adhere to recent Platform guidance to reduce the performance hit on such large queries. In my implementation I'm using a page size of 100 objects. My original implementation actually used `openPointInTimeForType` and `closePointInTime` directly, but during my draft reviews it was pointed out that the PIT finder is a much cleaner + better alternative, which I agree with. 

**Why I made the change to add the `searchBehavior` option:** because, in my (admittedly limited) manual testing of ES search behavior with sort and search_after, I was under the impression that wouldn't work for saved object attributes that are undefined and/or not unique. For my use case, there aren't any required fields that are also unique.

However, after playing with this some more, I realize that each hit's `sort` field contains _two_ elements, and this pair is always unique. Re-reading the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/paginate-search-results.html#search-after), it looks like this second element is a tiebreaker value, but I don't recall see that before in my manual testing. I guess I was searching without using a PIT, so the tiebreaker wasn't present 🙈.

TL;DR it seems like this change isn't really necessary since using PIT adds a tiebreaker for search values, and sort/searchAfter will work for my use case. In the interest of reducing complexity, I'll just rip out my additions unless you or someone else thinks it would be valuable to leave them in. WDYT?

Reply:
> it seems like this change isn't really necessary since using PIT adds a tiebreaker for search values, and sort/searchAfter will work for my use case

Thanks for the background! This is part of why I asked -- if you don't have a reason _not_ to use searchAfter and your only concern is a tiebreaker, then you should be all good and don't need to change anything here. 👌

ES [automatically applies a tiebreaker](https://www.elastic.co/guide/en/elasticsearch/reference/master/paginate-search-results.html#search-after) for you  any time you are searching with a pit and search_after, which is what you were seeing:

> All PIT search requests add an implicit sort tiebreaker field called _shard_doc, which can also be provided explicitly. If you cannot use a PIT, we recommend that you include a tiebreaker field in your sort. This tiebreaker field should contain a unique value for each document. If you don’t include a tiebreaker field, your paged results could miss or duplicate hits.

So I'm +1 on removing these changes if we don't have another need for them.